doc/api/stream.md

```diff
@@ -341,136 +553,140 @@ reader.on('end', () => {
 });
 ```
 
-Note that [`process.stderr`][] and [`process.stdout`][] are never closed until
-the process exits, regardless of the specified options.
+*Note*: The [`process.stderr`][] and [`process.stdout`][] Writable streams are
+never closed until the Node.js process exits, regardless of the specified
+options.
 
-#### readable.read([size])
+##### readable.read([size])
 
 * `size` {Number} Optional argument to specify how much data to read.
 * Return {String|Buffer|Null}
 
-The `read()` method pulls some data out of the internal buffer and
-returns it. If there is no data available, then it will return
-`null`.
+The `readable.read()` method pulls some data out of the internal buffer and
+returns it. If no data available to be read, `null` is returned. By default,
+the data will be returned as a `Buffer` object unless an encoding has been
+specified using the `readable.setEncoding()` method.
 
-If you pass in a `size` argument, then it will return that many
-bytes. If `size` bytes are not available, then it will return `null`,
-unless we've ended, in which case it will return the data remaining
-in the buffer.
+The optional `size` argument specifies a specific number of bytes to read. If
+`size` bytes are not available to be read, `null` will be returned *unless*
+the stream has ended, in which case all of the data remaining in the internal
+buffer will be returned (*even if it exceeds `size` bytes*).
 
-If you do not specify a `size` argument, then it will return all the
-data in the internal buffer.
+If the `size` argument is not specified, all of the data contained in the
+internal buffer will be returned.
 
-This method should only be called in paused mode. In flowing mode,
-this method is called automatically until the internal buffer is
-drained.
+The `readable.read()` method should only be called on Readable streams operating
+in paused mode. In flowing mode, `readable.read()` is called automatically until
+the internal buffer is fully drained.
 
 ```js
-var readable = getReadableStreamSomehow();
+const readable = getReadableStreamSomehow();
 readable.on('readable', () => {
   var chunk;
   while (null !== (chunk = readable.read())) {
-    console.log('got %d bytes of data', chunk.length);
+    console.log(`Received ${chunk.length} bytes of data.`);
   }
 });
 ```
```

Comment:
Again it should be noted that while this example is a totally viable way to obtain chunks from a stream, in practice it's best to use `.on('data')` or `.pipe()` and avoid `.read()` entirely.